xarray/core/accessor_dt.py

```diff
@@ -209,14 +220,12 @@ def __init__(self, obj: T_DataArray) -> None:
     def _date_field(self, name: str, dtype: DTypeLike) -> T_DataArray:
         if dtype is None:
             dtype = self._obj.dtype
-        obj_type = type(self._obj)
-        result = _get_date_field(self._obj.data, name, dtype)
-        return obj_type(result, name=name, coords=self._obj.coords, dims=self._obj.dims)
+        result = _get_date_field(_index_or_data(self._obj), name, dtype)
+        return self._obj.copy(data=result).rename(name)
```

Comment:
This will deep copy, is thatn intentional?
Maybe better to always specify the deep argument to prevent such accidents.

Reply:
I'm not sure we want that. 

I think we want `_replace`. Is that more like the previous version? Push a commit?

Reply:
Ah yes, replace is probably best here.
I'm on mobile only currently, so I will not be able to work on anything seriously...